core/pkg/eval/json_evaluator.go

```diff
@@ -57,6 +65,12 @@ func (je *JSONEvaluator) GetState() (string, error) {
 }
 
 func (je *JSONEvaluator) SetState(payload sync.DataSync) (map[string]interface{}, bool, error) {
+	_, span := je.jsonEvalTracer.Start(
+		context.Background(),
+		"setState",
+		trace.WithAttributes(attribute.String("source", payload.Source)))
```

Comment:
to avoid confusion with the [messaging semconv](https://github.com/open-telemetry/opentelemetry-specification/blob/25f513dbea2810be1c6f94d314d1a8d7e7b592c9/semantic_conventions/trace/messaging.yaml#L59), I suggest the following
```suggestion
		trace.WithAttributes(attribute.String("feature_flag.source", payload.Source)))
```